src/compiler/debug.rs

```diff
@@ -0,0 +1,24 @@
+#![allow(non_upper_case_globals)]
+
+pub struct DebugType {}
+
+impl DebugType {
+    fn fail(&self, message: Option<&str>) -> ! {
+        let message = match message {
+            Some(message) => format!("Debug failure. {}", message),
+            None => "Debug failure.".to_string(),
+        };
+        panic!("{}", message);
+    }
+
+    pub fn assert(&self, expression: bool) {
+        if !expression {
+            let message = "False expression.";
+            self.fail(Some(message));
+        }
+    }
+}
```

Comment:
Since the `DebugType` struct doesn't really do anything, I'd probably make these free functions that you call using the module name to qualify them:

```rust
debug::assert(false)
```